### PR TITLE
Update data-migration documentation

### PR DESCRIPTION
**Pull request by michplunkett**
## Fixes issue
https://github.com/lucyparsons/OpenOversight/issues/930

## Description of Changes
Add documentation to make data-migrations a more transparent process.

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -6,18 +6,18 @@ Babel~=2.12.1
 bleach==6.0.0
 bleach-allowlist==1.0.3
 blinker~=1.6.2
-boto3==1.28.1
-botocore==1.31.1
+boto3==1.28.3
+botocore==1.31.3
```
Comment (michplunkett) [inline requirements.txt:10]: Updated package versions since I was already in the file.

Comment (abandoned-prototype) [reply requirements.txt:10]: I would prefer to not mix things that have nothing to do with each other. Makes reverting things harder and also harder to cherry-pick for forked projects.

File: OpenOversight/migrations/alembic.ini [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 [alembic]
 # template used to generate migration files
-# file_template = %%(rev)s_%%(slug)s
+file_template = %%(year)d_%%(month).2d_%%(day).2d_%%(hour).2d%%(minute).2d-%%(rev)s_%%(slug)s
```
[the remark this answers is not in the document]
Comment (abandoned-prototype) [reply OpenOversight/migrations/alembic.ini:3]: This is a really great addition! I would prefer `%%(year)d-%%(month).2d-%%(day).2d-%%(hour).2d%%(minute).2d_%%(rev)s_%%(slug)s`

File: requirements.txt (hunk 1 above)
Comment (michplunkett) [reply requirements.txt:10]: Makes sense to me! I'll revert these changes.

File: OpenOversight/migrations/env.py [hunk 3]
```diff
@@ -62,7 +50,7 @@ def run_migrations_online():
     # this callback is used to prevent an auto-migration from being generated
     # when there are no changes to the schema
     # reference: http://alembic.zzzcomputing.com/en/latest/cookbook.html
-    def process_revision_directives(context, revision, directives):
```
Comment (michplunkett) [inline OpenOversight/migrations/env.py:65]: The first two parameters are not being used and are masking outside variables, so I removed them.

File: OpenOversight/migrations/README.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+## General Alembic Information
+Alembic provides for the creation, management, and invocation of change management scripts for a relational database, using SQLAlchemy as the underlying engine.
+
+For information on how to execute DB migrations, please visit this section in the `CONTRIB.md` file: [Link](../../CONTRIB.md#migrating-the-database)
```
Comment (michplunkett) [inline OpenOversight/migrations/README.md:4]: References the `Migrating the Database` part of the `CONTRIB.md` file.

**Review by b-meson — APPROVED** [review]

**Pull request merged**